### PR TITLE
Add verbiage to release notes about physical memory mgmt improvements

### PR DESCRIPTION
**Pull request by ghale**
Mention the improvements made in #23912.

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -233,6 +233,16 @@ See the [Test.forkEvery](dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.a
 
 When using the `init` task with the `--incubating` option, [parallel project execution](userguide/multi_project_configuration_and_execution.html#sec:parallel_execution) and [task output caching](userguide/build_cache.html) will be enabled for the generated project (by creating a `gradle.properties` file and setting the appropriate flags in it).
 
+### General Improvements
+
+#### Better physical memory management
+Gradle attempts to manage its physical memory usage by proactively stopping unused worker processes before starting up new ones.
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:239]: 🧐 
```suggestion
Gradle attempts to manage its physical memory usage by proactively stopping unused worker processes before starting new ones.  
```

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -233,6 +233,16 @@ See the [Test.forkEvery](dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.a
 
 When using the `init` task with the `--incubating` option, [parallel project execution](userguide/multi_project_configuration_and_execution.html#sec:parallel_execution) and [task output caching](userguide/build_cache.html) will be enabled for the generated project (by creating a `gradle.properties` file and setting the appropriate flags in it).
 
+### General Improvements
+
+#### Better physical memory management
+Gradle attempts to manage its physical memory usage by proactively stopping unused worker processes before starting up new ones.  
+It does this by first checking if the available physical memory can accommodate the heap requirements of a new worker process.
+If not, Gradle then looks for unused worker processes that can be stopped in order to free up enough physical memory for the new process.
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:241]: 🧐 
```suggestion
If not, Gradle then looks for unused worker processes that can be stopped to free up enough physical memory for the new process.
```

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -233,6 +233,16 @@ See the [Test.forkEvery](dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.a
...
+Previously, it sought to acquire enough memory to satisfy the minimum heap requirements of the new process, but in cases where the minimum heap and maximum heap of the worker process are very different, the memory freed up before the process starts may not be close to sufficient for the eventual size of the process.
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:242]: 🧐 
```suggestion
Previously, it sought to acquire enough memory to satisfy the minimum heap requirements of the new process. However, in cases where the minimum heap and maximum heap of the worker process are very different, the memory freed up before the process starts may not be close to sufficient for the eventual size of the process.  
```

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -233,6 +233,16 @@ See the [Test.forkEvery](dsl/org.gradle.api.tasks.testing.Test.html#org.gradle.a
...
+### General Improvements
+
+#### Better physical memory management
+Gradle attempts to manage its physical memory usage by proactively stopping unused worker processes before starting up new ones.  
+It does this by first checking if the available physical memory can accommodate the heap requirements of a new worker process.
+If not, Gradle then looks for unused worker processes that can be stopped in order to free up enough physical memory for the new process.
+Previously, it sought to acquire enough memory to satisfy the minimum heap requirements of the new process, but in cases where the minimum heap and maximum heap of the worker process are very different, the memory freed up before the process starts may not be close to sufficient for the eventual size of the process.  
+Gradle now attempts to acquire enough memory to satisfy the _maximum_ heap requirements of the new process.
+This causes the physical memory management to be more aggressive when starting up new processes, and in many cases will result in better overall memory usage.
+
```
Comment (hythloda) [inline subprojects/docs/src/docs/release/notes.md:245]: Is there a link to fine-tuning the heap arguments we can add here?

Comment (ghale) [reply subprojects/docs/src/docs/release/notes.md:247]: There's not much, but I've added a link to what currently exists.

**Comment by ghale** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by ghale** [comment]
@bot-gradle cancel

**Comment by ghale** [comment]
This is being added separately.

**Pull request closed without merging**